src/operator/numpy/np_matrix_op.cc

```diff
@@ -1151,17 +1151,14 @@ inline bool TrilindicesOpShape(const nnvm::NodeAttrs& attrs,
   int m = param.m;
   int k = param.k;
 
-  int length = 0;
+  index_t length = 0;
   int end = k;
   for (int i = 0; i < n; i++) {
-    int tmpCount = 0;
-    for (int j = 0; j <= std::min(end, m - 1); j++) {
-      tmpCount++;
-    }
-    length += tmpCount;
+    int mi = std::min(end, m - 1);
```

Comment:
Good !
you can remove `if` by 
```
int mi = std::max(std::min(end, m - 1), 0);
```
Looks clean IMO, upto you if you think if condition improves readability you can keep it